std/format.d

```diff
@@ -3040,19 +3040,46 @@ if (isInputRange!T)
             static if (is(StringTypeOf!T))
             {
                 auto s = val[0 .. f.precision < $ ? f.precision : $];
+
+                size_t width;
+                if (f.width > 0)
+                {
+                    // ASCII dstring & string can be aligned w/o graphemeStride
```

Comment:
Actually, that's wrong. `dstring` also needs grapheme cluster decoding, because a grapheme like `a\u0310\u0337` occupies multiple dchars but only takes up one space when displayed.

Reply:
No this is right, this works, but i can add a test for that `assert("%2s".format("a\u0310\u0337"d) == " a\u0310\u0337");`
i meant  _"dstring only made of ASCII and string only made of ASCII..."_. In this case we have 3 `dchars`. only 1 is ASCII so `onlyAscii` is false and then the execution branches to the stride code.

Reply:
you seem to forget that `dstring`s are decoded... ;)

Reply:
Yes, but why is the optimization not applied to `wstring`? If I have a wstring containing only ASCII characters, shouldn't it skip the grapheme stride code too?